src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -236,35 +236,33 @@ int fmi3_xml_get_attr_str(fmi3_xml_parser_context_t* context, fmi3_xml_elm_t elm
         return -1;
     }
 
-    *valp = value; /* can be NULL */
+    *valp = value; /* NULL if missing (and not required) */
     return 0;
 }
 
 /**
  * Reads the attribute from attribute buffer as jm_vector(char). This will clear the attribute from the buffer.
+ * @param exists (return arg): If attribute exists (to tell empty string vs missing attribute)
```

Comment:
Effectively unused in a lot of places, particularly if required = 1.

Still better than having 2 distinct functions I think.